pkg/s3select/input.go

```diff
@@ -288,20 +218,31 @@ type Row struct {
 // Execute is the function where all the blocking occurs, It writes to the HTTP
 // response writer in a streaming fashion so that the client can actively use
 // the results before the query is finally finished executing. The
-func (reader *Input) Execute(writer io.Writer) error {
+func Execute(writer io.Writer, inputType SelectQuery) error {
 	myRow := make(chan *Row)
 	curBuf := bytes.NewBuffer(make([]byte, 1000000))
 	curBuf.Reset()
 	progressTicker := time.NewTicker(progressTime)
 	continuationTimer := time.NewTimer(continuationTime)
 	defer progressTicker.Stop()
 	defer continuationTimer.Stop()
-	go reader.runSelectParser(convertMySQL(reader.options.Expression), myRow)
+	var expression string
+
+	switch inputType.(type) {
+	case *JSONInput:
+		expression = inputType.getExpression()
+
+	case *CSVInput:
+		expression = inputType.getExpression()
+	}
```

Comment:
Unknown input type lets return an error right here..

Reply:
Its already checked in object-handler ```if selectReq.InputSerialization.CompressionType == "" {
		selectReq.InputSerialization.CompressionType = s3select.SelectCompressionNONE
		if !strings.Contains(objInfo.ContentType, "text/csv") && !strings.Contains(objInfo.ContentType, "application/json") {
			writeErrorResponse(w, ErrInvalidDataSource, r.URL)
			return
		}
	}``` . 
Why do we need the check here again?

Reply:
Yes but since this is an Init for s3select package we can return an error.